hw-model/src/model_verilated.rs

```diff
@@ -111,10 +112,38 @@ fn ahb_txn_size(ty: AhbTxnType) -> RvSize {
         AhbTxnType::ReadU64 | AhbTxnType::WriteU64 => RvSize::Word,
     }
 }
+impl SocManager for ModelVerilated {
+    type TMmio<'a> = BusMmio<VerilatedApbBus<'a>>;
 
-impl crate::HwModel for ModelVerilated {
+    fn mmio_mut(&mut self) -> Self::TMmio<'_> {
+        BusMmio::new(self.apb_bus())
+    }
+
+    fn delay(&mut self) {
+        self.step();
+    }
+
+    const SOC_IFC_ADDR: u32 = 0x3003_0000;
```

Comment:
Might be nice to move these constants to somewhere shared just so they aren't repeated (and we are less likely to make a typo).